07-task-aliasing.Rmd

```diff
@@ -0,0 +1,80 @@
+```{r, include = FALSE}
+ottrpal::set_knitr_image_path()
+```
+
+# Calling the same task more than once 
+It is not uncommon that the same task (for example the BwaMem task in our example) in a workflow is required to be used multiple times. For example you may want to do this if you need to run it using a different set of inputs, if you want to split your task to scale your workflow and optimize compute resources. 
+
+WDL has a sophisticated feature that allows one to reuse the same task repeatedly through your workflow: **task aliasing**
+Simply put **task aliasing**  allows for the re-use of task definitions within the same workflow under different names, or "aliases". 
+
+This feature not only prevents redundancy but also significantly simplifies the workflow's maintenance. 
+
+By leveraging task aliasing, users can update a single task definition, and those changes will automatically apply wherever the task's alias is used, ensuring consistency across the workflow.
+
+The major advantages of using task aliasing are:
+
+ **1. Reduces Redundancy**: You don't need to copy and paste the same task definition multiple times and your workflows will be more concise and organized. 
+**2. Simplifies Maintenance** : If you decide to change/update/fix a task, using task aliasing will make life easy as you need to update only once in your workflow. 
+ **3. Enhances Readability and Clarity**: A shorter workflow is easier to read but task aliasing also helps to contextualize the workflow ( for example are you doing this task for Sample set A or Sample set B)
+**4. Facilitates Modular Workflow Design**: Task aliasing help to make your workflow modular. This is easier to adopt by **5. Improves Workflow Scalability**: Using task aliasing it is much easier to scale the workflow across different inputs. For example you want to run a task on different sample groups (Sample set A and B) will allow the same task to be run parallely and if you choose with different modifications. 
+**6. Ensures Consistency**: Task aliasing assures that there is consistency in replicated tasks and helps the reader easily identify where changes are expected in a task. 
```

Comment:
Make sure there's line breaks between each listed number for proper formatting when it's rendered.